In [1]:
import pandas as pd 

In [2]:
path_to_titan = "../data/TITAN_TestSet_global_scores/testSet_Global_scores_TITAN.csv"
df_titan = pd.read_csv(path_to_titan)
df_titan

TRA_nt  \
0      ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
1      ATGCTCCTGGAGCTTATCCCACTGCTGGGGATACATTTTGTCCTGA...   
2      ATGGCAGGCATTCGAGCTTTATTTATGTACTTGTGGCTGCAGCTGG...   
3      ATGAGGCAAGTGGCGAGAGTGATCGTGTTCCTGACCCTGAGTACTT...   
4      ATGGAGAAGAATCCTTTGGCAGCCCCATTACTAATCCTCTGGTTTC...   
...                                                  ...   
10518  ATGGCATGCCCTGGCTTCCTGTGGGCACTTGTGATCTCCACCTGTC...   
10519  ATGTGGGGAGCTTTCCTTCTCTATGTTTCCATGAAGATGGGAGGCA...   
10520  ATGATATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGTTAA...   
10521  ATGAACTATTCTCCAGGCTTAGTATCTCTGATACTCTTACTGCTTG...   
10522  ATGATATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGTTAA...   

                                                  TRB_nt  \
0      ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
1      ATGGGCCCCCAGCTCCTTGGCTATGTGGTCCTTTGCCTTCTAGGAG...   
2      ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
3      ATGGGCCCCGGGCTCCTCTGCTGGGCACTGCTTTGTCTCCTGGGAG...   
4      ATGGGCACCAGTCTCCTATGCTGGGTGGTCCTGGGTTTCCTAGGGA...   
...                                                  ...   
10518  ATGCTGCTGCTTCTGCTGCTTCTGGGGCCAGGTATAAGCCTCCTTC...   
10519  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
10520  ATGGGCTCCAGGCTGCTCTGTTGGGTGCTGCTTTGTCTCCTGGGAG...   
10521  ATGGGCCCTGGGCTCCTCTGCTGGGTGCTGCTTTGTCTCCTGGGAG...   
10522  ATGCTGAGTCTTCTGCTCCTTCTCCTGGGACTAGGCTCTGTGTTCA...   

                                                  TRA_aa  \
0      MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
1      MLLELIPLLGIHFVLRTARAQSVTQPDIHITVSEGASLELRCNYSY...   
2      MAGIRALFMYLWLQLDWVSRGESVGLHLPTLSVQEGDNSIINCAYS...   
3      MRQVARVIVFLTLSTLSLAKTTQPISMDSYEGQEVNITCSHNNIAT...   
4      MEKNPLAAPLLILWFHLDCVSSILNVEQSPQSLHVQEGDSTNFTCS...   
...                                                  ...   
10518  MACPGFLWALVISTCLEFSMAQTVTQSQPEMSVQEAETVTLSCTYD...   
10519  MWGAFLLYVSMKMGGTAGQSLEQPSEVTAVEGAIVQINCTYQTSGF...   
10520  MISLRVLLVILWLQLSWVWSQRKEVEQDPGPFNVPEGATVAFNCTY...   
10521  MNYSPGLVSLILLLLGRTRGNSVTQMEGPVTLSEEAFLTINCTYTA...   
10522  MISLRVLLVILWLQLSWVWSQRKEVEQDPGPFNVPEGATVAFNCTY...   

                                                  TRB_aa             TRAV  \
0      MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...      TRAV12-3*01   
1      MGPQLLGYVVLCLLGAGPLEAQVTQNPRYLITVTGKKLTVTCSQNM...       TRAV8-3*01   
2      MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...      TRAV13-2*01   
3      MGPGLLCWALLCLLGAGLVDAGVTQSPTHLIKTRGQQVTLRCSPKS...         TRAV4*01   
4      MGTSLLCWVVLGFLGTDHTGAGVSQSPRYKVTKRGQDVALRCDPIS...        TRAV24*01   
...                                                  ...              ...   
10518  MLLLLLLLGPGISLLLPGSLAGSGLGAVVSQHPSWVICKSGTSVKI...  TRAV38-2/DV8*01   
10519  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...       TRAV1-1*01   
10520  MGSRLLCWVLLCLLGAGPVKAGVTQTPRYLIKTRGQQVTLSCSPIS...      TRAV12-1*01   
10521  MGPGLLCWVLLCLLGAGSVETGVTQSPTHLIKTRGQQVTLRCSSQS...       TRAV9-2*01   
10522  MLSLLLLLLGLGSVFSAVISQKPSRDICQRGTSLTIQCQVDSQVTM...      TRAV12-1*01   

            TRAJ         TRA_CDR3         TRBV        TRBJ           TRB_CDR3  \
0      TRAJ42*01  CAMGGSGGSQGNLIF    TRBV19*01  TRBJ2-1*01      CASSIRASNEQFF   
1      TRAJ49*01      CAWATGNQFYF    TRBV27*01  TRBJ2-1*01   CASSFSGGALYNEQFF   
2      TRAJ42*01  CAENLGGGSQGNLIF    TRBV19*01  TRBJ2-1*01      CASSPRSGGEQFF   
3       TRAJ4*01  CLVASPSGGYNKLIF   TRBV5-6*01  TRBJ2-5*01     CASSLARQEETQYF   
4      TRAJ15*01     CALNQAGTALIF   TRBV7-6*01  TRBJ1-1*01    CASSLAQQGLAEAFF   
...          ...              ...          ...         ...                ...   
10518  TRAJ43*01     CAYRFNNNDMRF  TRBV20-1*01  TRBJ1-1*01     CSATKDRGYTEAFF   
10519  TRAJ33*01      CAVNSNYQLIW    TRBV19*01  TRBJ1-2*01  CASSKPGTGLSGGGYTF   
10520  TRAJ43*01      CVVNNNNDMRF   TRBV5-1*01  TRBJ2-2*01      CASNEENTGELFF   
10521   TRAJ6*01     CALFAGSYIPTF   TRBV5-4*01  TRBJ2-5*01    CASSSRTGDQETQYF   
10522  TRAJ17*01     CVVGAAGNKLTF  TRBV29-1*01  TRBJ2-2*01  CSASGPTGGRD

In [3]:

print(f"columns of titan df: \n{df_titan.columns}")
first_row_titan = df_titan.iloc[0]
first_row_titan

columns of titan df: 
Index(['TRA_nt', 'TRB_nt', 'TRA_aa', 'TRB_aa', 'TRAV', 'TRAJ', 'TRA_CDR3',
       'TRBV', 'TRBJ', 'TRB_CDR3', 'TRAC', 'TRBC', 'TRA_leader', 'A1', 'A2',
       'A3', 'B1', 'B2', 'B3', 'score', 'epitope'],
      dtype='object')


TRA_nt        ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...
TRB_nt        ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...
TRA_aa        MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...
TRB_aa        MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...
TRAV                                                TRAV12-3*01
TRAJ                                                  TRAJ42*01
TRA_CDR3                                        CAMGGSGGSQGNLIF
TRBV                                                  TRBV19*01
TRBJ                                                 TRBJ2-1*01
TRB_CDR3                                          CASSIRASNEQFF
TRAC                                                    TRAC*01
TRBC                                                   TRBC2*01
TRA_leader                                       TRAV12-3*01(L)
A1                                                       NSAFQY
A2                                                       TYSSGN
A3                                      

In [4]:
first_row_titan["TRA_CDR3"]

'CAMGGSGGSQGNLIF'

In [5]:
path_to_VDJdb = "../data/VDJdb/VDJdb_data.tsv"
df_vdjdb = pd.read_csv(path_to_VDJdb, sep="\t")
df_vdjdb

complex.id Gene                  CDR3            V           J  \
0               1  TRB  CASSYLPGQGDHYSNQPQHF    TRBV13*01  TRBJ1-5*01   
1               0  TRB   CASSFEAGQGFFSNQPQHF    TRBV13*01  TRBJ1-5*01   
2               2  TRB   CASSFEPGQGFYSNQPQHF    TRBV13*01  TRBJ1-5*01   
3               3  TRB  CASSYEPGQVSHYSNQPQHF    TRBV13*01  TRBJ1-5*01   
4               4  TRB        CASSALASLNEQFF    TRBV14*01  TRBJ2-1*01   
...           ...  ...                   ...          ...         ...   
84180       30590  TRA         CMDEGGSNYKLTF  TRAV26-1*01   TRAJ53*01   
84181       30591  TRA           CSLYNNNDMRF  TRAV26-1*01   TRAJ43*01   
84182       30592  TRA         CALSTDSWGKLQF     TRAV6*01   TRAJ24*01   
84183       30593  TRA          CAPQGATNKLIF  TRAV12-2*01   TRAJ32*01   
84184       30594  TRA        CLVGGSGGYNKLIF     TRAV4*01    TRAJ4*01   

           Species           MHC A           MHC B MHC class        Epitope  \
0      HomoSapiens        HLA-B*08             B2M      MHCI       FLKEKGGL   
1      HomoSapiens        HLA-B*08             B2M      MHCI       FLKEKGGL   
2      HomoSapiens        HLA-B*08             B2M      MHCI       FLKEKGGL   
3      HomoSapiens        HLA-B*08             B2M      MHCI       FLKEKGGL   
4      HomoSapiens        HLA-B*08             B2M      MHCI       FLKEKGGL   
...            ...             ...             ...       ...            ...   
84180  HomoSapiens  HLA-DQA1*05:01  HLA-DQB1*02:01     MHCII   PQPELPYPQPQL   
84181  HomoSapiens  HLA-DQA1*05:01  HLA-DQB1*02:01     MHCII   PQPELPYPQPQL   
84182  HomoSapiens  HLA-DQA1*05:01  HLA-DQB1*02:01     MHCII  PQQPFPQPEQPFP   
84183  HomoSapiens  HLA-DQA1*05:01  HLA-DQB1*02:01     MHCII  PQQPFPQPEQPFP   
84184  HomoSapiens  HLA-DQA1*05:01  HLA-DQB1*02:01     MHCII  PQQPFPQPEQPFP   

      Epitope gene Epitope species      Reference  \
0              Nef           HIV-1  PMID:15596521   
1              Nef           HIV-1  PMID:15596521   
2              Nef           HIV-1  PMID:15596521   
3              Nef           HIV-1  PMID:15596521   
4              Nef           HIV-1  PMID:15596521   
...            ...             ...            ...   
84180       Gluten           Wheat  PMID:33927715   
84181       Gluten           Wheat  PMID:33927715   
84182       Gluten           Wheat  PMID:33927715   
84183       Gluten           Wheat  PMID:33927715   
84184       Gluten           Wheat  PMID:33927715   

                                                  Method  \
0      {"frequency": "", "identification": "tetramer-...   
1      {"frequency": "", "identification": "tetramer-...   
2      {"frequency": "", "identification": "tetramer-...   
3      {"frequency": "", "identification": "tetramer-...   
4      {"frequency": "", "identification": "tetramer-...   
...                                                  ...   
84180  {"frequency": "", "identification": "tetramer-...   
84181  {"frequency": "", "identification": "tetramer-...   
84182  {"frequency": "", "identification": "tetramer-...   
84183  {"frequency": "", "identification": "tetramer-...   
84184  {"frequency": "", "identification": "tetramer-...   

                                                    Meta  \
0      {"cell.subset": "CD8+", "clone.id": "", "donor...   
1      {"cell.subset": "CD8+", "clone.id": "", "donor...   
2      {"cell.subset": "CD8+", "clone.id": "", "donor...   
3      {"cell.subset": "CD8+", "clone.id": "", "donor...   
4      {"cell.subset": "CD8+", "clone.id": "", "donor...   
...                                                  ...   
84180  {"cell.subset": "CD4+", "clone.id": "", "donor...   
84181  {"cell.subset": "CD4+", "clone.id": "", "donor...   
84182  {"cell.subset": "CD4+", "clone.id": "", "donor...   
84183  {"cell.subset": "CD4+", "clone.id": "", "donor...   
84184  {"cell.subset": "CD4+", "clone.id": "", "donor...   

                                                 CDR3fix  Score  
0      {"cdr3": "CASSYLPGQGDHYSNQPQH

In [6]:
print(first_row_titan["TRA_CDR3"])
print(first_row_titan["TRB_CDR3"])

CAMGGSGGSQGNLIF
CASSIRASNEQFF


In [7]:
dummy = df_vdjdb[(df_vdjdb["CDR3"] == first_row_titan["TRA_CDR3"]) | (df_vdjdb["CDR3"] == first_row_titan["TRB_CDR3"])]
dummy

complex.id Gene             CDR3            V           J      Species  \
9524            0  TRB    CASSIRASNEQFF    TRBV19*01  TRBJ2-1*01  HomoSapiens   
13063        1261  TRB    CASSIRASNEQFF    TRBV19*01  TRBJ2-1*01  HomoSapiens   
14430           0  TRB    CASSIRASNEQFF    TRBV19*01  TRBJ2-1*01  HomoSapiens   
55128        1261  TRA  CAMGGSGGSQGNLIF  TRAV12-3*01   TRAJ42*01  HomoSapiens   

             MHC A MHC B MHC class    Epitope Epitope gene Epitope species  \
9524      HLA-A*02   B2M      MHCI  GILGFVFTL            M      InfluenzaA   
13063  HLA-A*02:01   B2M      MHCI  GILGFVFTL            M      InfluenzaA   
14430  HLA-A*02:01   B2M      MHCI  GILGFVFTL            M      InfluenzaA   
55128  HLA-A*02:01   B2M      MHCI  GILGFVFTL            M      InfluenzaA   

           Reference                                             Method  \
9524   PMID:28423320  {"frequency": "0.0349%", "identification": "an...   
13063  PMID:28636589  {"frequency": "11/321", "identification": "tet...   
14430  PMID:28629751  {"frequency": "2.8%", "identification": "tetra...   
55128  PMID:28636589  {"frequency": "11/321", "identification": "tet...   

                                                    Meta  \
9524   {"cell.subset": "CD8+", "clone.id": "", "donor...   
13063  {"cell.subset": "", "clone.id": "", "donor.MHC...   
14430  {"cell.subset": "CD8+", "clone.id": "", "donor...   
55128  {"cell.subset": "", "clone.id": "", "donor.MHC...   

                                                 CDR3fix  Score  
9524   {"cdr3": "CASSIRASNEQFF", "cdr3_old": "CASSIRA...      0  
13063  {"cdr3": "CASSIRASNEQFF", "cdr3_old": "CASSIRA...      0  
14430  {"cdr3": "CASSIRASNEQFF", "cdr3_old": "CASSIRA...      2  
55128  {"cdr3": "CAMGGSGGSQGNLIF", "cdr3_old": "CAMGG...      0

This is weird... but I (Cyril) do not trust the data from the TITAN specifically this file. What I (Cyril) expected were that we could find somewhere in the VDJdb the entries of the alpha and the beta chain. The alpha-chain 55128 and the beta-chain(s) would fit but the epitope is wrong. Score is very low and we do not know that this means => i assume that this is a synthetically created "false positive" example!!

In [8]:
'''
from IPython.display import display, HTML
# Convert the DataFrame to HTML
df_html = df.to_html()

# Create a scrollable table using custom HTML and CSS
scrollable_table = f"""
<div style="overflow-x: auto; width: 800px; height: 200px; overflow-y: scroll;">
    {df_html}
</div>
"""

# Display the scrollable table
display(HTML(scrollable_table))
'''

'\nfrom IPython.display import display, HTML\n# Convert the DataFrame to HTML\ndf_html = df.to_html()\n\n# Create a scrollable table using custom HTML and CSS\nscrollable_table = f"""\n<div style="overflow-x: auto; width: 800px; height: 200px; overflow-y: scroll;">\n    {df_html}\n</div>\n"""\n\n# Display the scrollable table\ndisplay(HTML(scrollable_table))\n'

We used the Stitchr command line tool first to be able to reproduce one TCRA_nt, TCRB_nt, TCRA_aa and TCRB_aa 

For this, we use the information of the first paired chain data of the VDJdb.

**Input: (the TRB)**

`stitchr -v TRBV19*01 -j TRBJ2-1*01 -cdr3 CASSIRASNEQFF`

**Output:** 

TRB_nt: 

`ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAGCAAACACCGTGGAT
GGTGGAATCACTCAGTCCCCAAAGTACCTGTTCAGAAAGGAAGGACAGAATGTGACCCTG
AGTTGTGAACAGAATTTGAACCACGATGCCATGTACTGGTACCGACAGGACCCAGGGCAA
GGGCTGAGATTGATCTACTACTCACAGATAGTAAATGACTTTCAGAAAGGAGATATAGCT
GAAGGGTACAGCGTCTCTCGGGAGAAGAAGGAATCCTTTCCTCTCACTGTGACATCGGCC
CAAAAGAACCCGACAGCTTTCTATCTCTGTGCCAGTAGTATCAGAGCCAGCAATGAGCAG
TTCTTCGGGCCAGGGACACGGCTCACCGTGCTAGAGGACCTGAAAAACGTGTTCCCACCC
GAGGTCGCTGTGTTTGAGCCATCAGAAGCAGAGATCTCCCACACCCAAAAGGCCACACTG
GTGTGCCTGGCCACAGGCTTCTACCCCGACCACGTGGAGCTGAGCTGGTGGGTGAATGGG
AAGGAGGTGCACAGTGGGGTCAGCACAGACCCGCAGCCCCTCAAGGAGCAGCCCGCCCTC
AATGACTCCAGATACTGCCTGAGCAGCCGCCTGAGGGTCTCGGCCACCTTCTGGCAGAAC
CCCCGCAACCACTTCCGCTGTCAAGTCCAGTTCTACGGGCTCTCGGAGAATGACGAGTGG
ACCCAGGATAGGGCCAAACCTGTCACCCAGATCGTCAGCGCCGAGGCCTGGGGTAGAGCA
GACTGTGGCTTCACCTCCGAGTCTTACCAGCAAGGGGTCCTGTCTGCCACCATCCTCTAT
GAGATCTTGCTAGGGAAGGCCACCTTGTATGCCGTGCTGGTCAGTGCCCTCGTGCTGATG
GCCATGGTCAAGAGAAAGGATTCCAGAGGC`

--------------------------------------------------------------------------------

TRB_aa: 

`MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNLNHDAMYWYRQDPGQ
GLRLIYYSQIVNDFQKGDIAEGYSVSREKKESFPLTVTSAQKNPTAFYLCASSIRASNEQ
FFGPGTRLTVLEDLKNVFPPEVAVFEPSEAEISHTQKATLVCLATGFYPDHVELSWWVNG
KEVHSGVSTDPQPLKEQPALNDSRYCLSSRLRVSATFWQNPRNHFRCQVQFYGLSENDEW
TQDRAKPVTQIVSAEAWGRADCGFTSESYQQGVLSATILYEILLGKATLYAVLVSALVLM
AMVKRKDSRG`

--------------------------------------------------------------------------------
--------------------------------------------------------------------------------

**Input: (the TRA)**

`stitchr -v TRAV12-3*01 -j TRAJ42*01 -cdr3 CAMGGSGGSQGNLIF`

**Output:** 

TRA_nt: 

`ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGTTAAGCTGGGTTTGG
AGCCAACAGAAGGAGGTGGAGCAGGATCCTGGACCACTCAGTGTTCCAGAGGGAGCCATT
GTTTCTCTCAACTGCACTTACAGCAACAGTGCTTTTCAATACTTCATGTGGTACAGACAG
TATTCCAGAAAAGGCCCTGAGTTGCTGATGTACACATACTCCAGTGGTAACAAAGAAGAT
GGAAGGTTTACAGCACAGGTCGATAAATCCAGCAAGTATATCTCCTTGTTCATCAGAGAC
TCACAGCCCAGTGATTCAGCCACCTACCTCTGTGCAATGGGCGGCAGCGGAGGAAGCCAA
GGAAATCTCATCTTTGGAAAAGGCACTAAACTCTCTGTTAAACCAAATATCCAGAACCCT
GACCCTGCCGTGTACCAGCTGAGAGACTCTAAATCCAGTGACAAGTCTGTCTGCCTATTC
ACCGATTTTGATTCTCAAACAAATGTGTCACAAAGTAAGGATTCTGATGTGTATATCACA
GACAAAACTGTGCTAGACATGAGGTCTATGGACTTCAAGAGCAACAGTGCTGTGGCCTGG
AGCAACAAATCTGACTTTGCATGTGCAAACGCCTTCAACAACAGCATTATTCCAGAAGAC
ACCTTCTTCCCCAGCCCAGAAAGTTCCTGTGATGTCAAGCTGGTCGAGAAAAGCTTTGAA
ACAGATACGAACCTAAACTTTCAAAACCTGTCAGTGATTGGGTTCCGAATCCTCCTCCTG
AAAGTGGCCGGGTTTAATCTGCTCATGACGCTGCGGCTGTGGTCCAGC`

--------------------------------------------------------------------------------

TRA_aa: 

`MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCTYSNSAFQYFMWYRQ
YSRKGPELLMYTYSSGNKEDGRFTAQVDKSSKYISLFIRDSQPSDSATYLCAMGGSGGSQ
GNLIFGKGTKLSVKPNIQNPDPAVYQLRDSKSSDKSVCLFTDFDSQTNVSQSKDSDVYIT
DKTVLDMRSMDFKSNSAVAWSNKSDFACANAFNNSIIPEDTFFPSPESSCDVKLVEKSFE
TDTNLNFQNLSVIGFRILLLKVAGFNLLMTLRLWSS`

In [9]:
TRB_nt_expected = "ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAGCAAACACCGTGGAT\
GGTGGAATCACTCAGTCCCCAAAGTACCTGTTCAGAAAGGAAGGACAGAATGTGACCCTG\
AGTTGTGAACAGAATTTGAACCACGATGCCATGTACTGGTACCGACAGGACCCAGGGCAA\
GGGCTGAGATTGATCTACTACTCACAGATAGTAAATGACTTTCAGAAAGGAGATATAGCT\
GAAGGGTACAGCGTCTCTCGGGAGAAGAAGGAATCCTTTCCTCTCACTGTGACATCGGCC\
CAAAAGAACCCGACAGCTTTCTATCTCTGTGCCAGTAGTATCAGAGCCAGCAATGAGCAG\
TTCTTCGGGCCAGGGACACGGCTCACCGTGCTAGAGGACCTGAAAAACGTGTTCCCACCC\
GAGGTCGCTGTGTTTGAGCCATCAGAAGCAGAGATCTCCCACACCCAAAAGGCCACACTG\
GTGTGCCTGGCCACAGGCTTCTACCCCGACCACGTGGAGCTGAGCTGGTGGGTGAATGGG\
AAGGAGGTGCACAGTGGGGTCAGCACAGACCCGCAGCCCCTCAAGGAGCAGCCCGCCCTC\
AATGACTCCAGATACTGCCTGAGCAGCCGCCTGAGGGTCTCGGCCACCTTCTGGCAGAAC\
CCCCGCAACCACTTCCGCTGTCAAGTCCAGTTCTACGGGCTCTCGGAGAATGACGAGTGG\
ACCCAGGATAGGGCCAAACCTGTCACCCAGATCGTCAGCGCCGAGGCCTGGGGTAGAGCA\
GACTGTGGCTTCACCTCCGAGTCTTACCAGCAAGGGGTCCTGTCTGCCACCATCCTCTAT\
GAGATCTTGCTAGGGAAGGCCACCTTGTATGCCGTGCTGGTCAGTGCCCTCGTGCTGATG\
GCCATGGTCAAGAGAAAGGATTCCAGAGGC"

TRB_nt_expected

'ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAGCAAACACCGTGGATGGTGGAATCACTCAGTCCCCAAAGTACCTGTTCAGAAAGGAAGGACAGAATGTGACCCTGAGTTGTGAACAGAATTTGAACCACGATGCCATGTACTGGTACCGACAGGACCCAGGGCAAGGGCTGAGATTGATCTACTACTCACAGATAGTAAATGACTTTCAGAAAGGAGATATAGCTGAAGGGTACAGCGTCTCTCGGGAGAAGAAGGAATCCTTTCCTCTCACTGTGACATCGGCCCAAAAGAACCCGACAGCTTTCTATCTCTGTGCCAGTAGTATCAGAGCCAGCAATGAGCAGTTCTTCGGGCCAGGGACACGGCTCACCGTGCTAGAGGACCTGAAAAACGTGTTCCCACCCGAGGTCGCTGTGTTTGAGCCATCAGAAGCAGAGATCTCCCACACCCAAAAGGCCACACTGGTGTGCCTGGCCACAGGCTTCTACCCCGACCACGTGGAGCTGAGCTGGTGGGTGAATGGGAAGGAGGTGCACAGTGGGGTCAGCACAGACCCGCAGCCCCTCAAGGAGCAGCCCGCCCTCAATGACTCCAGATACTGCCTGAGCAGCCGCCTGAGGGTCTCGGCCACCTTCTGGCAGAACCCCCGCAACCACTTCCGCTGTCAAGTCCAGTTCTACGGGCTCTCGGAGAATGACGAGTGGACCCAGGATAGGGCCAAACCTGTCACCCAGATCGTCAGCGCCGAGGCCTGGGGTAGAGCAGACTGTGGCTTCACCTCCGAGTCTTACCAGCAAGGGGTCCTGTCTGCCACCATCCTCTATGAGATCTTGCTAGGGAAGGCCACCTTGTATGCCGTGCTGGTCAGTGCCCTCGTGCTGATGGCCATGGTCAAGAGAAAGGATTCCAGAGGC'

In [10]:
filtered_df= df_titan.filter(items=["TRA_nt", "TRB_nt", "TRA_aa", "TRB_aa"])

In [11]:
filtered_df = df_titan[df_titan['TRB_nt'] == TRB_nt_expected]
filtered_df

TRA_nt  \
0     ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
619   ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
1238  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
1857  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
2476  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
3095  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
3714  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
4333  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
4952  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
5571  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
6190  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
6809  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
7428  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
8047  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
8666  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
9285  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   
9904  ATGATGAAATCCTTGAGAGTTTTACTGGTGATCCTGTGGCTTCAGT...   

                                                 TRB_nt  \
0     ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
619   ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
1238  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
1857  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
2476  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
3095  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
3714  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
4333  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
4952  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
5571  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
6190  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
6809  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
7428  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
8047  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
8666  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
9285  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   
9904  ATGAGCAACCAGGTGCTCTGCTGTGTGGTCCTTTGTTTCCTGGGAG...   

                                                 TRA_aa  \
0     MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
619   MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
1238  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
1857  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
2476  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
3095  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
3714  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
4333  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
4952  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
5571  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
6190  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
6809  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
7428  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
8047  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
8666  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
9285  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   
9904  MMKSLRVLLVILWLQLSWVWSQQKEVEQDPGPLSVPEGAIVSLNCT...   

                                                 TRB_aa         TRAV  \
0     MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
619   MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
1238  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
1857  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
2476  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
3095  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
3714  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
4333  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
4952  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
5571  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01   
6190  MSNQVLCCVVLCFLGANTVDGGITQSPKYLFRKEGQNVTLSCEQNL...  TRAV12-3*01